packages/protocol/contracts/governance/DowntimeSlasher.sol

```diff
@@ -35,88 +42,212 @@ contract DowntimeSlasher is SlasherUtil {
   }
 
   /**
-   * @notice Sets the slashable downtime
+   * @notice Sets the slashable downtime.
    * @param interval Slashable downtime in blocks.
    */
   function setSlashableDowntime(uint256 interval) public onlyOwner {
-    require(interval != 0, "Slashable downtime cannot be zero");
-    require(interval < getEpochSize(), "Slashable downtime must be smaller than epoch size");
+    require(interval != 0, "slashable downtime cannot be zero");
+    require(interval < getEpochSize(), "slashable downtime must be smaller than epoch size");
     slashableDowntime = interval;
     emit SlashableDowntimeSet(interval);
   }
 
   /**
-   * @notice Test if a validator has been down.
-   * @param startBlock First block of the downtime. Last block will be computed from this.
-   * @param startSignerIndex Index of the signer within the validator set as of the start block.
-   * @param endSignerIndex Index of the signer within the validator set as of the end block.
-   * @return True if the validator signature does not appear in any block within the window.
-   * @dev Due to getParentSealBitmap, startBlock must be within 4 epochs of the current head.
+   * @notice Calculates and returns the signature bitmap for the specified interval.
+   * Similar to the parentSealBitmap of every block (where you have which validators were
```

Comment:
```
* @notice Calculates and returns the signature bitmap for the specified interval.
* This bitmap will contain a one for any validator that signed at least one block in that interval, and zero otherwise.
* @param ...
```